### DISCUSSION TITLE
how to display a list of tags, which are not mainTags

### DISCUSSION THREAD

**Discussion by 3muchfun** [GetPublii/Publii#1504] [Help] [answered]
Dear all, 
I use the code below and I would like to change it to display tags which are not mainTags, only. 
Maybe one of you can help me out...

```
{{#checkIfAny @config.post.displayTags @config.post.displayShareButtons }}
    {{#if @config.post.displayTags}}
        {{#if tags}}
            {{#each tags}}
                {{#checkIf @index '>' 0}}, {{/checkIf}}
                <a href="{{url}}">{{name}}</a>
            {{/each}}
        {{/if}}
    {{/if}}
{{/checkIfAny}}

**Comment by kryptot7** [comment]
I haven't tested this code, but just by taking a quick look, I'm not sure why the `{{checkIf}}` statements are next to each other, separated by a comma. I tried to fix the formatting so it compiles correctly.
I also got rid of an unnecessary space in the top statement.
Does this help?
````handlebars
{{#checkIfAny @config.post.displayTags @config.post.displayShareButtons}}
    {{#if @config.post.displayTags}}
        {{#if tags}}
            {{#each tags}}
                {{#checkIf @index '>' 0}}
                    <a href="{{url}}">{{name}}</a>
                {{/checkIf}}
            {{/each}}
        {{/if}}
    {{/if}}
{{/checkIfAny}}
````

**Reply by 3muchfun** [reply]
Thanks for your answer, @kryptot7.
The code I mentioned was provided by @bobmitro [here](https://github.com/GetPublii/Publii/discussions/1487#discussioncomment-9179037) and works fine. Secondly, I am missing the " , " in your code to create the comma separated list.
My question is, how can I filter the tags to show mainTags, only.

**Comment by 3muchfun** [comment]
@candidexmedia, do you have any idea on how to achieve that?

**Reply by candidexmedia** [reply]
Not sure if you still need help with this, but because mainTag is still new, and the dev docs have some gaps, it's hard to know what syntax to use. I'm also not a pro at handlebars / JS.

My hunch would be to add some sort of "not equal" argument, kind of like what's explained for [checkIf](https://getpublii.com/dev/checkif-helper/). Maybe @HowToMeetLadies has some ideas?

Sidenote for Publii devs & community: I would love to see some kind of crowd-sourced, searchable database/repo of code snippets for theme developpers, beyond the docs.

**Reply by candidexmedia** [reply]
Wanted to add: the Handlebars documentation might also have some clues: 

https://handlebarsjs.com/guide/builtin-helpers.html#if

https://handlebarsjs.com/guide/builtin-helpers.html#each

**Reply by candidexmedia** [reply]
Adding the note for reference for a solution in the meantime @3muchfun : https://github.com/GetPublii/Publii/issues/1329#issuecomment-2143431201

**Comment by dziudek** [comment]
In Publii v.0.46 there will be `isMainTag` flag available inside tag items in `tags`:
https://github.com/GetPublii/Publii/commit/bca2160f275ca5e1c22d4bc23e7714a1463686dc

**Reply by candidexmedia** [reply]
Sweet!

So the syntax would be something like this?

```hbs
{{#checkIf tags.isMaintag = false}}
...
{{/checkIf}}
```

**Reply by dziudek** [reply]
@candidexmedia even simpler:

```
{{#unless isMainTag}}
...
{{/unless}}
```

**Reply by 3muchfun** [reply]
Thanks for your answers, @candidexmedia , @dziudek . 
So, with this new "isMainTag", I can differ between tags and MainTags?
At the bottom of each post I want to show the Category aka MainTag, and in a second row, the Tags, aka the tags which are not MainTags.

**Reply by michaelfluegge** [reply]
Ok, I See what you mean, but I am not deep into that, unfortunately. My guess is, that it has to do with one of these variables, but ...   https://getpublii.com/dev/introduction-global-variables/
Hope, someone else is able to help you. @candidexmedia ?

**Comment by michaelfluegge** [comment, marked as answer]
Just to bring this topic to an end, this is my solution:

```<p class="post-categories">Categories:
    {{#if @config.post.displayTags}}
        {{#if mainTag}}
            <a href="{{mainTag.url}}">{{mainTag.name}}</a>
        {{/if}}
    {{/if}}
</p>
```

and 

```<p class="post-tags">
    <span>Tags:</span>
        {{#checkIfAny @config.post.displayTags }}
            {{#if @config.post.displayTags}}
                {{#if tags}}
                    {{#each tags}}
                        {{#checkIf @index '>' 0}}, {{/checkIf}}
                            <a href="{{url}}">{{name}}</a>
                    {{/each}}
                {{/if}}
            {{/if}}
        {{/checkIfAny}}
</p>
```

which brings this
<img width="438" height="240" alt="Bildschirmfoto 2025-09-17 um 09 16 01" src="https://github.com/user-attachments/assets/6540a493-3315-44f5-a8b1-493f90ba1838" />

into something like this: 
<img width="660" height="175" alt="Bildschirmfoto 2025-09-17 um 09 15 28" src="https://github.com/user-attachments/assets/77bcff25-0127-44e3-b5db-46ad296d1590" />


CC: @candidexmedia

**Reply by candidexmedia** [reply]
Thanks for sharing your solution!

Did you have any luck with this to filter out the non mainTags?

```
{{#unless isMainTag}}
...
{{/unless}}
```

**Reply by michaelfluegge** [reply]
I tried it but with no success. But maybe I am not Pro enough ...

**Reply by candidexmedia** [reply]
Hi @dziudek : Could you advise on how to properly use `{{#unless isMainTag}}` ?

I've tried the following, which hasn't worked:

```hbs
  {{#if tags}}
     <ul class="content__tag">
        {{#each tags}}
           {{#unless isMainTag}}                                   
              <li>
                 <a href="{{url}}">{{name}}</a>
              </li>
           {{/unless}}
        {{/each}}
     </ul>
  {{/if}}
```

**Reply by bobmitro** [reply]
Use it this way: 

```
{{#if tags}}
  <ul class="content__tag">
    {{#each tags}}
      {{#if ../mainTag}}
        {{#checkIf this.id '!=' ../mainTag.id}}
          <li>
            <a href="{{url}}">{{name}}</a>
          </li>
        {{/checkIf}}
      {{else}}
        <li>
          <a href="{{url}}">{{name}}</a>
        </li>
      {{/if}}
    {{/each}}
  </ul>
{{/if}}
```

**Reply by candidexmedia** [reply]
Thank you so much @bobmitro!

@3muchfun : Here's what your updated code would look like:

```hbs
{{#checkIfAny @config.post.displayTags }}
      {{#if @config.post.displayTags}}
         {{#if tags}}
            {{#each tags}}
               {{#if ../mainTag}}
                  {{#checkIf this.id '!=' ../mainTag.id}}
                     {{#checkIf @index '>' 0}}, {{/checkIf}}
                        <a href="{{url}}">{{name}}</a>
                  {{/checkIf}}
               {{/if}}
            {{/each}}
         {{/if}}
      {{/if}}
{{/checkIfAny}}
```
